src/mount/readdata.cc

```diff
@@ -583,6 +656,18 @@ int read_to_buffer(ReadRecord *rrec, uint64_t current_offset,
 			if (size_in_chunk > bytes_to_read) {
 				size_in_chunk = bytes_to_read;
 			}
+
+			uint32_t read_cache_bytes_to_reserve =
+				getBytesToBeReadFromCS(reader.index(), offset_in_chunk,
+										size_in_chunk, reader.fileLength());
+			std::unique_lock lock(gReadCacheMemoryMutex);
+			bool successfulWait =
+			    waitForEnoughSystemMemory(read_cache_bytes_to_reserve);
+			if (!successfulWait) {
+				break;
+			}
```

Comment:
After the break, SAUNAFS_STATUS_OK will be returned, which is not correct.
Following the rest of the function logic, looks like we need to throw an exception, and then the logic of retries is executed.
According to this, we need to evaluate if the new retry logic in the waitForEnoughSystemMemory is really needed.
Sorry, I did not saw it in my first review.

Reply:
Done, integrated corresponding wait result with retry logic already implemented in code.